packages/core-mobile/app/store/account/listeners.ts

```diff
@@ -71,69 +73,61 @@ const initAccounts = async (
     accounts[acc.index] = { ...acc, name: accountTitle }
 
     listenerApi.dispatch(setAccounts(accounts))
-    if (isDeveloperMode === false) {
-      AnalyticsService.captureWithEncryption('AccountAddressesUpdated', {
-        addresses: Object.values(accounts).map(account => ({
-          address: account.addressC,
-          addressBtc: account.addressBTC,
-          addressAVM: account.addressAVM ?? '',
-          addressPVM: account.addressPVM ?? '',
-          addressCoreEth: account.addressCoreEth ?? ''
-        }))
-      })
-    }
+  }
+
+  if (isDeveloperMode === false) {
+    AnalyticsService.captureWithEncryption('AccountAddressesUpdated', {
+      addresses: Object.values(accounts).map(acc => ({
+        address: acc.addressC,
+        addressBtc: acc.addressBTC,
+        addressAVM: acc.addressAVM ?? '',
+        addressPVM: acc.addressPVM ?? '',
+        addressCoreEth: acc.addressCoreEth ?? ''
+      }))
+    })
   }
 }
 
-const fetchingRemainingAccounts = async ({
-  isDeveloperMode,
+const fetchRemainingAccounts = async ({
   walletType,
   activeAccountIndex,
-  listenerApi,
-  initialAccounts
+  startIndex,
+  listenerApi
 }: {
-  isDeveloperMode: boolean
   walletType: WalletType
   activeAccountIndex: number
+  startIndex: number
   listenerApi: AppListenerEffectAPI
-  initialAccounts: AccountCollection
-}): Promise<void> => {
+}): Promise<AccountCollection> => {
   /**
    * note:
    * adding accounts cannot be parallelized, they need to be added one-by-one.
    * otherwise race conditions occur and addresses get mixed up.
    */
   const state = listenerApi.getState()
   const activeNetwork = selectActiveNetwork(state)
-  const pubKeysStorage = new SeedlessPubKeysStorage()
-  const pubKeys = await pubKeysStorage.retrieve()
+  const pubKeys = await SeedlessPubKeysStorage.retrieve()
+  const numberOfAccounts = pubKeys.filter(pubKey =>
+    pubKey.derivationPath.startsWith(EVM_BASE_DERIVATION_PATH_PREFIX)
+  ).length
```

Comment:
looks like we are doing this in a few places. should we create a util for the filter logic? pubKeys.filter(isEvmPubkey)?